### ISSUE TITLE
Infinite loop when crate is built in a watched directory

### ISSUE THREAD

**Issue by doomnoodles** [rwasm/vite-plugin-rsw#24]
If outDir directory is in a directory that is watched for changes(in this example the watched directory is src), then the files in outDir should not trigger dependency change, but currently they do and this creates an infinite loop of compilation. Not a critical bug but was pretty annoying to figure out.

![image](https://user-images.githubusercontent.com/91634342/135347542-bd91f45c-17f2-48cf-92de-37ea2b63956e.png)
![image](https://user-images.githubusercontent.com/91634342/135347592-3ebda4b9-de1e-4827-947d-834b3da41d79.png)

**Comment by lencx** [comment]
Thanks for the report，and it will be supported by adding `exclude` in the next version. https://github.com/lencx/vite-plugin-rsw/issues/22#issuecomment-929752914

[RswPluginOptions](https://github.com/lencx/vite-plugin-rsw/blob/bb95373d3b/src/types.ts#L26-L28)
```ts
// Plugin options
export type RswPluginOptions = BaseRswConfig & {
  crates: Array<string|RswCrateOptions>;
  // Stop watching files, directories, or glob patterns, takes an array of strings
  exclude?: string[];
}
```

**Comment by lencx** [comment]
The infinite loop has been fixed in `v1.10.0`, now you can try.

**Comment by cfac45** [comment]
> The infinite loop has been fixed in `v1.10.0`, now you can try.

My example works now with this config:
```javascript
ViteRsw({ 
	root: "src/lib/",
	crates: [{ name: "example-rust-crate-rs" }],
	unwatch: ["*/pkg/*"]
}) 
```
Thanks for the great plugin!

**Comment by cfac45** [comment]
Does not work:
```javascript
ViteRsw({ 
	root: "src/lib/",
	crates: [{ name: "example-rust-crate-rs", unwatch: './pkg' }],
}) 
```

```
[rsw::npm::link]
  ↳ example-rust-crate-rs   C:\Users\USER\Documents\Development\sveltekit-rust-ssr-template\src\lib\example-rust-crate-rs\pkg

> i.map is not a function
TypeError: i.map is not a function
    at C:\Users\USER\Documents\Development\sveltekit-rust-ssr-template\node_modules\vite-plugin-rsw\dist\index.js:235:426
    at Array.forEach (<anonymous>)
    at ue (C:\Users\USER\Documents\Development\sveltekit-rust-ssr-template\node_modules\vite-plugin-rsw\dist\index.js:235:320)
    at Object.configureServer (C:\Users\USER\Documents\Development\sveltekit-rust-ssr-template\node_modules\vite-plugin-rsw\dist\index.js:236:929)  
    at Object.createServer (C:\Users\USER\Documents\Development\sveltekit-rust-ssr-template\node_modules\vite\dist\node\chunks\dep-f5e099f1.js:66734:41)
    at async Watcher.init_server (file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/chunks/index.js:4340:15)
    at async Watcher.init (file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/chunks/index.js:4247:3)
    at async file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/cli.js:836:20
file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/chunks/index.js:4390
                        throw new Error('Cannot close server before it is initialized');
                              ^

Error: Cannot close server before it is initialized
    at Watcher.close (file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/chunks/index.js:4390:10)
    at process.<anonymous> (file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/chunks/index.js:4235:9)
    at process.emit (node:events:365:28)
    at process.emit (node:domain:470:12)
    at process.exit (node:internal/process/per_thread:189:15)
    at handle_error (file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/cli.js:798:10)
    at file:///C:/Users/USER/Documents/Development/sveltekit-rust-ssr-template/node_modules/@sveltejs/kit/dist/cli.js:856:4
```
